Bump @storybook/react major version

vcanales opened the pull request:
<!-- Learn the overall process and best practices for pull requests at https://github.com/WordPress/gutenberg/blob/master/docs/contributors/repository-management.md#pull-requests. -->

## Description
This fixes some breakage related to dependency imports that we can see now [on the playground](https://wordpress.github.io/gutenberg/?path=/story/*):

<img width="718" alt="Screen Shot 2020-09-08 at 13 26 26" src="https://user-images.githubusercontent.com/1157901/92503154-3d7baf00-f1d7-11ea-9a31-4e70848ed948.png">

Looking at `moment-timezone` the `undefined is not an object (evaluating 'root.moment')` error means that for some reason Storybook is building the dependencies for the browser, and not passing `window` into the context of the function. I think that Storybook fixed this import/build issue as a part of their new major version :)

## How has this been tested?
I was able to reproduce the error locally, and it's fixed when updating Storybook.

## Types of changes
Bump `@storybook/react`, which was actually reporting that an upgrade was available.

vcanales committed:
Bump storybook major version

Fixes some breakage related to dependency imports.

mkaz reviewed (approved):
Tested it out and upgrade seems to have fixed the issue.
I didn't notice any issues with the stories either.

Thanks 👍

vcanales committed:
Increase buffer size on check licenses script

The size of the output for `npm ls --json --long --prod` has grown to
around 110MB. For this reason, the output was being truncated at the
previous maximum buffer size, which was set at around ~104MB.

vcanales commented:
The License Check test was failing because the output of  the `npm ls --json --long --dev` command grew over the maxBuffer setting of 104MB, to around `110MB`. I increased  the max buffer by a few megabytes to fix and give us a little bit of leeway. In the long run, we might run into this again, particularly because we don't know how our dependencies treat their respective dependencies, so we might want to reconsider how we define this buffer size limit.

mkaz commented:
Good catch @vcanales Thanks!
